diff --git a/upd02_get_manifests_from_updates.py b/upd02_get_manifests_from_updates.py
index f69c6f1..a287413 100644
--- a/upd02_get_manifests_from_updates.py
+++ b/upd02_get_manifests_from_updates.py
@@ -23,7 +23,7 @@ class UpdateNotSupported(Exception):
 
 
 def get_update_download_urls(download_uuid):
-    url = f'https://uupdump.net/json-api/get.php?id={download_uuid}&lang=en-us&edition=professional'
+    url = f'https://uupdump.net/json-api/get.php?id={download_uuid}'
     r = requests.get(url)
 
     if r.status_code == 500:
@@ -58,9 +58,18 @@ def get_update_download_urls(download_uuid):
         ]:
             continue
 
-        extension = Path(file).suffix.lower()
-        if extension not in ['.cab', '.esd', '.psf', '.msu', '.wim']:
-            raise Exception(f'Unknown file extension: {extension}')
+        if (
+            'Microsoft-Windows-Client-LanguagePack-Package'.lower() in file.lower()
+            and file.lower().endswith('.esd')
+        ):
+            pass
+        elif (
+            'LanguageExperiencePack'.lower() in file.lower()
+            and file.lower().endswith('.appx')
+        ):
+            pass
+        else:
+            continue
 
         urls.append({
             'name': file,
@@ -270,8 +279,11 @@ def extract_update_files(local_dir: Path):
 
     # Extract ESD files.
     esd_files = list(local_dir.glob('*.esd'))
+    esd_files += list(local_dir.glob('*.appx'))
     for esd_file in esd_files:
-        extract_dir = local_dir.joinpath(f'_extract_{next_extract_dir_num}')
+        # extract_dir = local_dir.joinpath(f'_extract_{next_extract_dir_num}')
+        extract_dir = local_dir.joinpath(f'{esd_file.stem}')
+        assert not extract_dir.exists()
         print(f'Extracting {esd_file} to {extract_dir}')
         next_extract_dir_num += 1
 
@@ -307,6 +319,8 @@ def extract_update_files(local_dir: Path):
             cab_extract(cab_file, extract_dir)
             cab_file.unlink()
 
+    input('Press Enter to continue...')
+
     # Starting with Windows 11, manifest files are compressed with the DCM v1
     # format. Use SXSEXP to de-compress them: https://github.com/hfiref0x/SXSEXP
     #
